pkg/proto/nbcontract/v1/nodepoolprofile.proto

```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message NodePoolProfile {
```

Comment:
not sure if this is the best name, and whether we need a separate proto for it? do we expect to have more fields here? if not, then vm size can just be in the main message of the contract

Reply:
@anujmaheshwari1 There will be more fields in the nodepoolprofile. But since they are not part of the GPUConfig, I didn't put them there now. The original name from AKS-RP is Agentpoolprofile. I just think NodePool is easier to understand to me. Or you think Agentpool is a more official name?

Reply:
Agentpool is a term very specific to AKS, I think we should make this specific to a node, rather than a particular bootstrapper, what do you think?

Reply:
As we synced, let's start with NodePoolProfile and see if we can come up with a better name

Reply:
I agree with looking hard for opportunities to get rid of the message. Hard to judge without knowing which other fields may come in there. But having vm_size under NodePoolProfile is already confusing, since we are starting to support mixed SKU NodePools.

Reply:
Regarding NodePool vs AgentPool - I think both terms are AKS-specific. NodePool is more descriptive though